andela/Titan-Devs
# #163518688  User Can Create Profile

#### What does this PR do?
Enables the functionality of updating, fetching, and deleting the user profile. 
#### Description of Task to be completed?
* Create update profile functionality
* Create fetch user profile functionality 
* Create delete user profile functionality 
* Create tests for profile creation 
#### How should this be manually tested?
* Pull the remote branch using `git pull origin ft-profile-creation-163518688` 
* `cd project_directory`
* Run the database migrations by `sequelize db:migrate`
* Run tests using `yarn test` or
* Start the server using `yarn start`
* Sign up to the app by consuming `POST: /api/v1/users`
* Login to the app by consuming `POST: /api/v1/users/login`
* Grab the token and put in headers of the request like `{Authorization: Bearer <token> }`as you consume `PUT: /api/v1/profiles/:username` to update the profile. The body of the request should look like `req.body: { profile:{
	                               username:"YvesIraguha",
	                                 firstname:"yvesIraguhawuara",
	                                 bio: "I am a javascript developer"
                                          }
                                     }`

  

#### Any background context you want to provide?
After a user has signed up, he/she might want to update the profile. Therefore, these endpoints are for editing the profile created when a user signs up. 
#### What are the relevant pivotal tracker stories?
163518688
